src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -879,7 +879,7 @@ private AuthenticationTicket ValidateToken(string idToken, OpenIdConnectMessage
             return ticket;
         }
 
-        private async Task ValidateOpenIdConnectProtocolAsync(JwtSecurityToken jwt, OpenIdConnectMessage message, bool ValidateCHash = true)
+        private async Task ValidateOpenIdConnectProtocolAsync(JwtSecurityToken jwt, OpenIdConnectMessage message)
```

Comment:
We should be calling this validation before firing the SecurityTokenValidated notification, so that users will not perform any work on an invalid message.